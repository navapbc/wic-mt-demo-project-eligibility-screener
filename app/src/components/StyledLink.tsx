import React from "react";
import Link from 'next/link'

export interface StyledLinkProps {
  href: string
  text: string
  external?: boolean
}

export const StyledLink: React.FC<StyledLinkProps> = ({href, text, external = false}) => {
  let className = 'usa-link'
  if (external) {
    className += ' usa-link--external'
  }

  return (
    <Link href={href} passHref>
      <a className={className}>{text}</a>
    </Link>
  )
}

export default StyledLink
