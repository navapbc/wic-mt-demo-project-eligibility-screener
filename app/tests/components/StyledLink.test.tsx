import StyledLink from '@components/StyledLink'

import { testSnapshot } from '../helpers/sharedTests'

const testProps = {
  href: '/somewhere',
  textKey: 'link text',
  external: false,
}

it('should match snapshot for internal link', () => {
  const element = <StyledLink {...testProps} />
  testSnapshot(element)
})

it('should match snapshot for external link', () => {
  const element = <StyledLink {...testProps} external={true} />
  testSnapshot(element)
})
