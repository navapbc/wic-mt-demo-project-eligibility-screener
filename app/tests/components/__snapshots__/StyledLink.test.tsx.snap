// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for external link 1`] = `
<a
  className="usa-link usa-link--external"
  href="/somewhere"
  rel="noopener noreferrer"
  target="_blank"
>
  link text
</a>
`;

exports[`should match snapshot for internal link 1`] = `
<a
  className="usa-link"
  href="/somewhere"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  link text
</a>
`;
